scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
-        commodity_dict[x] = timeseries
+    # get list of sum(quantity) and qualid for snf
+    snf_inventory = cursor.execute(exec_string(sink_id,
+                                               'transactions.receiverId',
+                                               'sum(quantity), qualid') +
+                                   ' GROUP BY qualid').fetchall()
+    compositions = cursor.execute('SELECT qualid, nucid, massfrac '
+                                  'FROM compositions').fetchall()
+    snf_dict = collections.defaultdict(float)
+    for comp in compositions:
+        for num in snf_inventory:
+            if num[0] == comp[0]:
+                snf_dict[comp[1]] += num[1] * comp[2]
 
-    return commodity_dict
+    return snf_dict
 
 
-def get_from_facility(cursor, facility, duration, resources):
-    """ Returns timeseries list of quantity out of facility type
+def get_isotope_transactions(resources, compositions):
+    """Creates a dictionary with isotope name, mass, and time
 
     Parameters
     ----------
-    cursor: sqlite cursor
-        sqlite cursor
-    facility: str
-        name of facility type
-    duration: int
-        duration of simulation
-    resrources: list
-        list of lists, of quantity, senderid and time
+    resources: list
+        resource data from the resources table
+        list[0]: time
+        list[1]: sum(quantity)
+        list[2]: qualid
+    compositions: list
+        composition data from the compositions table
+        list[0]: qualid
+        list[1]: nucid
+        list[2]: massfrac
 
     Returns
     -------
-    list
-    timeseries list of mass outflux from facility type
+    transactions: dictionary
+        dictionary with keys as isotope and value as
+        list of tuples (mass moved, time)
     """
+    transactions = collections.defaultdict(list)
+    for res in resources:
+        for comp in compositions:
+            # res_qualid = res[2]
+            # comp_qualid = comp[0]
+            if res[2] == comp[0]:
+                # comp_nucid = comp[1]
+                # res_quantity = res[1s
+                # mass_frac = comp[2]
+                # res_time = res[0]
+                transactions[comp[1]].append((res[0], res[1] * comp[2]))
 
-    quantity = 0
-    timeseries = []
-    agentid = get_agent_id_from_prototype(cursor, facility)
-    for i in range(0, duration):
-        indices = [x for x, y in enumerate(resources) if y[2] == i]
-        for index in indices:
-            try:
-                if str(resources[index][1]) in agentid:
-                    quantity += resources[index][0]
-            except:
-                print('none in this timestep' + str(i))
-        timeseries.append(quantity/1000)
-
-    return timeseries
+    return transactions
 
 
-def total_waste_timeseries(cursor):
-    """Plots a stacked bar chart of the total waste mass vs time
+def commodity_in_out_facility(cursor, facility, commod_list,
```

Comment:
I've decided to rename the function to: facility_commodity_flux and facility_commodity_flux_isotopics.
After discussing with Teddy, we've adopted the following changes:
 - Split the function to "do_isotopic" function and "no_isotopics" function
 - Remove the parameters: facility, and is_prototype and replace it with agent_ids

We couldn't find a way to split this function even further, as each line performs basic tasks such as appending a dictionary, making a sqlite query. We'll keep looking, though!